seed/python-sdk/validation/with-defaults/src/seed/client.py

```diff
@@ -0,0 +1,325 @@
+
+    follow_redirects : typing.Optional[bool]
+        Whether the default httpx client follows redirects or not, this is irrelevant if a custom httpx client is passed in.
+
+    httpx_client : typing.Optional[httpx.Client]
+        The httpx client to use for making requests, a preconfigured client is used by default, however this is useful should you want to pass in any custom httpx configuration.
+
+    Examples
+    --------
+    from seed.client import SeedValidation
+
+    client = SeedValidation(
+        base_url="https://yourhost.com/path/to/api",
+    )
+    """
+
+    def __init__(
+        self,
+        *,
+        base_url: str,
+        timeout: typing.Optional[float] = None,
+        follow_redirects: typing.Optional[bool] = True,
+        httpx_client: typing.Optional[httpx.Client] = None
+    ):
+        _defaulted_timeout = timeout if timeout is not None else 60 if httpx_client is None else None
+        self._client_wrapper = SyncClientWrapper(
+            base_url=base_url,
+            httpx_client=httpx_client
+            if httpx_client is not None
+            else httpx.Client(timeout=_defaulted_timeout, follow_redirects=follow_redirects)
+            if follow_redirects is not None
+            else httpx.Client(timeout=_defaulted_timeout),
+            timeout=_defaulted_timeout,
+        )
+
+    def create(
+        self,
+        *,
+        decimal: float = 1,
+        even: int = 42,
+        name: str = "fern",
+        request_options: typing.Optional[RequestOptions] = None
+    ) -> Type:
+        """
+        Parameters
+        ----------
+        decimal : float
+
+        even : int
+
+        name : str
+
+        request_options : typing.Optional[RequestOptions]
+            Request-specific configuration.
+
+        Returns
+        -------
+        Type
+
+        Examples
+        --------
+        from seed.client import SeedValidation
+
+        client = SeedValidation(
+            base_url="https://yourhost.com/path/to/api",
+        )
+        client.create(
+            decimal=1.1,
+            even=1,
+            name="string",
+        )
+        """
+        _response = self._client_wrapper.httpx_client.request(
+            "create",
+            method="POST",
+            json={"decimal": decimal, "even": even, "name": name},
+            request_options=request_options,
+            omit=OMIT,
+        )
+        try:
+            if 200 <= _response.status_code < 300:
+                return pydantic_v1.parse_obj_as(Type, _response.json())  # type: ignore
+            _response_json = _response.json()
+        except JSONDecodeError:
+            raise ApiError(status_code=_response.status_code, body=_response.text)
+        raise ApiError(status_code=_response.status_code, body=_response_json)
+
+    def get(
+        self,
+        *,
+        decimal: float = 1,
+        even: int = 42,
+        name: str = "fern",
```

Comment:
let's say the default for a field typed as optional<integer> is 4, can the user still send `None` ? Is it worth adding an explicit test for that in the test definition

Reply:
I'll add a test
But yeah pydantic will prefer user provided over default